core/src/main/java/org/apache/accumulo/core/clientImpl/TabletServerBatchWriter.java

```diff
@@ -160,8 +165,9 @@ public class TabletServerBatchWriter implements AutoCloseable {
   private final HashSet<String> serverSideErrors = new HashSet<>();
   private final FailedMutations failedMutations;
   private int unknownErrors = 0;
-  private boolean somethingFailed = false;
+  private volatile boolean somethingFailed = false;
```

Comment:
This doesn't need to be changed in this PR, but I think there's a good case to be made to prefer the Java 5 atomic classes over manually making variables volatile.

* Atomic classes force correct usage
* Usually the extra object creation overhead doesn't matter (but obviously, don't use them when it does matter)
* final Atomic references allow the object's contained value to be updated inside a lambda or anonymous class
* One can immediately infer the memory characteristics by merely looking at the type of the variable in the IDE, and not needing to reference any initializing keywords, which makes for less IDE juggling back and forth, and doesn't require the developer to juggle that information in their head about which fields are declared volatile and which aren't.

Reply:
I try to always declare Atomics as final, so there's no additional concern about pointer chasing. But yeah, I get what you mean if that's not certain. Not really relevant here, but I remembered another good use of Atomics that you don't get directly with volatile: compareAndSet.